### PR TITLE
Add support for Code Climate issues

### LINKED ISSUE #571: Code Climate issue count
Code Climate has three sorts of badges offered on their website: GPA, test coverage and issue count. There are currently only shields for the first two of those; there doesn't appear to be any shield integration for the issue count.

As an example, here's the issue count for a small project of mine, as served by Code Climate from `https://codeclimate.com/github/me-and/mdf/badges/issue_count.svg`:

[Issue Count](https://codeclimate.com/github/me-and/mdf)

### PR DESCRIPTION
**Pull request by me-and**
Resolves #571.

I don't have the wherewithal to test this right now, so I'm hoping having just emulated existing code will be sufficient...

### CODE REVIEW

**Comment by me-and** [comment]
On the off chance anyone has taken a copy of this PR already, my apologies: I've just forced a push to my development branch to correct the email addresses used in the commits.

File: index.html [hunk 1]
```diff
@@ -523,6 +523,10 @@ <h3 id="miscellaneous"> Miscellaneous </h3>
     <td><img src='https://img.shields.io/codeclimate/coverage/github/triAGENS/ashikawa-core.svg' alt=''/></td>
     <td><code>https://img.shields.io/codeclimate/coverage/github/triAGENS/ashikawa-core.svg</code></td>
   </tr>
+  <tr><th> Code Climate: </th>
+    <td><img src='https://img.shields.io/codeclimate/issues/github/me-and/mdf.svg' alt=''/></td>
+    <td><code>https://img.shields.io/codeclimate/issues/github/me-and/mdf.svg</code></td>
+  </tr>
```
Comment (espadrine) [inline index.html]: index.html is automatically generated. Could you move that modification to try.html instead, please?

File: server.js [hunk 2]
```diff
@@ -2021,6 +2021,51 @@ cache(function(data, match, sendBadge, request) {
   });
 }));
 
+// Code Climate issues integration
+camp.route(/^\/codeclimate\/issues\/(.+)\.(svg|png|gif|jpg|json)$/,
+cache(function(data, match, sendBadge, request) {
+  var userRepo = match[1];  // eg, `github/me-and/mdf`.
+  var format = match[2];
+  var options = 'https://codeclimate.com/' + userRepo + 'badges/issue_count.svg'
```
Comment (espadrine) [inline server.js]: You are missing a `/` before `'badges/issue_count.svg'`.
Also, you are missing semicolon at the end of the

File: server.js [hunk 3]
```diff
@@ -2021,6 +2021,51 @@ cache(function(data, match, sendBadge, request) {
...
+  var badgeData = getBadgeData('code climate', data);
```
Comment (espadrine) [inline server.js]: Brands tend not to give much information about the values. Put "issues" instead of "code climate".

File: server.js [hunk 4]
```diff
@@ -2021,6 +2021,51 @@ cache(function(data, match, sendBadge, request) {
...
+        badgeData.text[1] = 'malformed';
+        sendBadge(format, badgeData);
+        return;
+      }
+      if (count == 1) {
+        badgeData.text[1] = '1 issue';
+      } else {
+        badgeData.text[1] = count + ' issues';
+      }
+      badgeData.text[1] = count;
```
Comment (espadrine) [inline server.js]: That line overrides the work done just above. I believe you want to remove the if / else.

File: server.js [hunk 5]
```diff
@@ -2021,6 +2021,51 @@ cache(function(data, match, sendBadge, request) {
...
+      } else if (count < 10) {
+        badgeData.colorscheme = 'yellowgreen';
+      } else if (count < 20) {
+        badgeData.colorscheme = 'yellow';
+      } else {
+        badgeData.colorscheme = 'red';
+      }
+      sendBadge(format, badgeData);
+    } catch(e) {
+      badgeData.text[1] = 'not found';
```
Comment (espadrine) [inline server.js]: It isn't quite not found in this case, I think. Perhaps 'invalid'?

**Comment by espadrine** [comment]
Thanks for contributing. Could you squash your commits together, please?

**Comment by me-and** [comment]
Sure! For some reason I didn't get any notifications for your replies. Thanks for the feedback; I'll get things fixed up and push an update.

**Comment by me-and** [comment]
Here's a new version, that I believe covers all the comments.

File: try.html [hunk 6]
```diff
@@ -558,6 +558,10 @@ <h3 id="miscellaneous"> Miscellaneous </h3>
     <td><img src='/codeclimate/coverage/github/triAGENS/ashikawa-core.svg' alt=''/></td>
     <td><code>https://img.shields.io/codeclimate/coverage/github/triAGENS/ashikawa-core.svg</code></td>
   </tr>
+  <tr><th> Code Climate: </th>
+    <td><img src='https://img.shields.io/codeclimate/issues/github/me-and/mdf.svg' alt=''/></td>
```
Comment (espadrine) [inline try.html]: Could you remove the `https://img.shields.io` part in the URL, please?

File: server.js [hunk 7]
```diff
@@ -2149,6 +2149,50 @@ cache(function(data, match, sendBadge, request) {
...
+      var count = buffer.match(/>([0-9]+) issues?/)[1];
+      if (!count) {
+        badgeData.text[1] = 'malformed';
+        sendBadge(format, badgeData);
+        return;
+      }
+      if (count == 1) {
+        badgeData.text[1] = '1 issue';
+      } else {
+        badgeData.text[1] = count + ' issues';
```
Comment (espadrine) [inline server.js]: Could you remove the `issue` part, please? I'd like the badge to just have a number on the right-hand side. Something like that:

**Comment by me-and** [comment]
Both fixed in the latest version.

**Comment by espadrine** [comment]
Thanks a lot for the contribution! I'll try to put this live within 24h.

**Pull request merged**